CONTRIBUTING.md

```diff
@@ -0,0 +1,140 @@
+  ```
+
+- From your GitHub interface, you should now be able to open a Pull Request to the JAX-GalSim repository. 
+
+Before submitting your PR, have a look at the procedure documented below.
+
+### Checklist before opening a Pull Request
+
+- Pull Requests should be self-contained and limited in scope, otherwise they become too difficult to review. If your modifications are broad, consider opening several smaller Pull Requests.
+
+- Make sure your fork and branch are up-to-date with the `main` branch of JAX-GalSim. To update your local branch, you may do so from the GitHub interface, or you may use this CLI command:
+  ```bash
+  # Only needs to be done once:
+  git remote add upstream http://www.github.com/GalSim-developers/JAX-GalSim
+  # This will update your local branch
+  git fetch upstream
+  git rebase upstream/main
+  ```
+
+- Make sure the unit tests still work:
+  ```bash
+  pytest
+  ```
+  Ideally there should be some new unit tests for the new functionality, unless the work is completely covered by existing unit tests.
+
+- Make sure your code conforms to the [Black](https://github.com/psf/black) style:
+  ```bash
+  black .
+  ```
+  If you haven't installed it already, you can install Black with `pip install black`.
+
+- Update CHANGELOG.md to mention your change.
+
+- Make sure any new files have BSD license at the top.
+
+- If your changes contain multiple commits, we encourage you to squash them into a single (or very few) commit, before opening the PR. To do so, you can using this command:
+```bash
+git rebase -i
+```
+
+### Opening the Pull Request
+
+- On the GitHub site, go to "Code". Then click the green "Compare and Review" button. Your branch is probably in the "Example Comparisons" list, so click on it. If not, select it for the "compare" branch.
+
+- Make sure you are comparing your new branch to the upstream `main`. Press Create Pull Request button.
+
+- Give a brief title. (We usually leave the branch number as the start of the title.)
+
+- Explain the major changes you are asking to be code reviewed. Often it is useful to open a second tab in your browser where you can look through the diff yourself to remind yourself of all the changes you have made.
+
+### After submitting the pull request
+
+- Check to make sure that the PR can be merged cleanly. If it can, GitHub will report that "This branch has no conflicts with the base branch." If it doesn't, then you need to merge from master into your branch and resolve any conflicts.
+
+- Wait a few minutes for the continuous integration tests to be run. Then make sure that the tests reports no errors. If not, click through to the details and try to figure out what is causing the error and fix it.
+
+### After code review
+
+- Once at least 1 and preferably 2 people have reviewed the code, and you have responded to all of their comments, we generally solicit for "any other comments" and give people a few more days before merging.
```

Comment:
For this to work, you need to have a few people committed to reviewing basically every PR.  For GalSim, that's currently me, Josh and Rachel.  Before Josh joined, Barney was the third.  I'm pretty sure there hasn't been a PR in GalSim that wasn't reviewed by one of us, plus sometimes one or two other interested people.  So you need to make sure there are two or three people committed enough to this project and willing to take that time commitment to guarantee this principle.

Reply:
Yes, definitely, that's the difficult part. Code review is pretty much mandatory, no getting around that. The format I picture in steady state is 2 (hopefully 3) maintainers, who distribute the reviews to developers (people who have joined the dev team, i.e. who have already had at least one PR merged and agreed to continue to help), and who ultimately press the merge button. 
In the initial phase though, this probably requires to gather a few committed people as you say, to get the project to a point where more people can start contributing. Working on that :-)